typing/types.ml

```diff
@@ -300,6 +300,12 @@ and constructor_tag =
   | Cstr_extension of Path.t * bool     (* Extension constructor
                                            true if a constant false if a block*)
 
+let equal_tag t1 t2 = match (t1, t2) with
+  | Cstr_constant i1, Cstr_constant i2 -> i2 = i1
+  | Cstr_block i1, Cstr_block i2 -> i2 = i1
+  | Cstr_extension (path1, b1), Cstr_extension (path2, b2) -> path1 = path2 && b1 = b2
+  | _ -> false
+
```

Comment:
An alternative to omitting the last case is to expand only the left column and group the non-matching constructors together:

```ocaml
    K1 p, K1 q -> ...
  | K2 p, K2 q -> ...
    ...
  | Kn p, Kn q -> ...
  | (K1 _|K2 _|...|Kn _), _ -> false
```

This seems to result in slightly better lambda code, but I also find it easier to read.